java/cdk/src/main/java/sleeper/cdk/stack/CompactionStack.java

```diff
@@ -288,10 +294,80 @@ private void lambdaToFindCompactionJobsThatShouldBeCreated(CoreStacks coreStacks
         instanceProperties.set(COMPACTION_JOB_CREATION_CLOUDWATCH_RULE, rule.getRuleName());
     }
 
-    private void ecsClusterForCompactionTasks(CoreStacks coreStacks,
-                                              IBucket jarsBucket,
-                                              LambdaCode taskCreatorJar,
-                                              Queue compactionJobsQueue) {
+    private void lambdaToCreateCompactionJobsBatchedViaSQS(
+            CoreStacks coreStacks, IBucket jarsBucket, LambdaCode jobCreatorJar, Queue compactionJobsQueue) {
+
+        // Function to create compaction jobs
+        Map<String, String> environmentVariables = Utils.createDefaultEnvironment(instanceProperties);
+
+        String triggerFunctionName = Utils.truncateTo64Characters(String.join("-", "sleeper",
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "compaction-job-creation-trigger"));
+        String functionName = Utils.truncateTo64Characters(String.join("-", "sleeper",
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "compaction-job-creation-handler"));
+
+        IFunction triggerFunction = jobCreatorJar.buildFunction(this, "CompactionJobsCreationTrigger", builder -> builder
+                .functionName(triggerFunctionName)
+                .description("Create batches of tables and send requests to create compaction jobs for those batches")
+                .runtime(JAVA_11)
+                .memorySize(256)
+                .timeout(Duration.seconds(30))
+                .handler("sleeper.compaction.job.creation.lambda.CreateCompactionJobsTriggerLambda::handleRequest")
+                .environment(environmentVariables)
+                .reservedConcurrentExecutions(1)
+                .logGroup(createLambdaLogGroup(this, "CompactionJobsCreationTriggerLogGroup", functionName, instanceProperties)));
+
+        IFunction handlerFunction = jobCreatorJar.buildFunction(this, "CompactionJobsCreationHandler", builder -> builder
+                .functionName(functionName)
+                .description("Scan DynamoDB looking for files that need compacting and create appropriate job specs in DynamoDB")
+                .runtime(JAVA_11)
+                .memorySize(instanceProperties.getInt(COMPACTION_JOB_CREATION_LAMBDA_MEMORY_IN_MB))
+                .timeout(Duration.seconds(instanceProperties.getInt(COMPACTION_JOB_CREATION_LAMBDA_TIMEOUT_IN_SECONDS)))
+                .handler("sleeper.compaction.job.creation.lambda.CreateCompactionJobsSQSLambda::handleRequest")
+                .environment(environmentVariables)
+                .logGroup(createLambdaLogGroup(this, "CompactionJobsCreationHandlerLogGroup", functionName, instanceProperties)));
+
+        // Grant this function permission to read from / write to the DynamoDB table
```

Comment:
This comment doesn't seem right - can we have one comment which says all the things that need access, including the one below under "Grant this function permission to put messages on the compaction queue"?

Reply:
Done, thanks.